src/System.IO.FileSystem/src/System/IO/WinRTFileSystem.cs

```diff
@@ -227,14 +228,107 @@ public override bool DirectoryExists(string fullPath)
 
         public override IEnumerable<string> EnumeratePaths(string fullPath, string searchPattern, SearchOption searchOption, SearchTarget searchTarget)
         {
-            // Fill in implementation
-            return new string[0];
+            IEnumerable<IStorageItem> storageFiles = SynchronousResultOf(EnumerateFileQuery(fullPath, searchPattern, searchOption, searchTarget));
+
+            // Yield the paths of the StorageItems
+            foreach (IStorageItem item in storageFiles)
```

Comment:
Could the return type of `EnumerateFileQuery` be `Task<IReadOnlyList<IStorageItem>>` instead of `Task<IEnumerable<IStorageItem>>`?

This would allow indexing into the `IReadOnlyList<T>`, which I suspect is more efficient than using the enumerator  because it would avoid the enumerator allocation and any adapter/RCW allocations for the WinRT projection/interop, and instead of two calls for each item (e.g. `MoveNext()` and `get_Current`) it can just make one call to the indexer (although, I have not measured to confirm). e.g.

```csharp
int count = storageFiles.Count;
for (int i = 0; i < count; i++)
{
    yield return storageFiles[i].Path;
}
```